[Conversation] Add button to scroll to previous user message

philipperolet opened the pull request:
## Description

Adds an up-arrow button on user messages (similar to the existing scroll-to-bottom button) that scrolls to the previous user message in the conversation. This helps users navigate through their own messages in long conversations.

https://github.com/user-attachments/assets/ab5cb802-a2c7-405f-b0ca-2ab9742b71d4


The button:
- Only appears when there is a previous user message to scroll to
- Uses smooth scrolling for short distances, instant for longer ones (same threshold as scroll-to-bottom)
- Uses the existing `ArrowUpIcon` from sparkle
- 
## Risks

Blast radius: User message navigation in conversations
Risk: low

## Deploy Plan

- deploy front

philipperolet committed:
[Conversation] Add button to scroll to previous user message

Add an up-arrow button on user messages that scrolls to the previous
user message in the conversation. This helps users navigate through
their messages in long conversations.

philipperolet committed:
Show button only on hover and add tooltip

JulesBelveze reviewed (commented):
I like the idea but is it design approved ? 😅 I'm not convinced by the current one..

philipperolet committed:
Move navigation to bottom bar with up/down buttons

Replace the scroll-to-bottom button with up/down navigation buttons
that allow navigating between user messages. The buttons are properly
disabled when at the first/last user message and have tooltips.

philipperolet committed:
Always show navigation buttons, fix disabled states

- Show up/down buttons always (not just when scrolled away)
- When at bottom: up is active (can go to last user message), down is disabled
- When at first user message: up is disabled, down is active

philipperolet committed:
Fix navigation buttons and update design

- Fix reactivity issue by computing user message indices on demand
- Put both arrows in a single container with border (matching design)
- Use ghost variant for buttons inside the container

philipperolet committed:
Fix navigation issues: blank space, first click, scroll awareness

1. Fix blank space: When navigating to last user message, scroll to
   bottom ("LAST" with "end" align) instead of to message with "start"
2. Fix first click: From bottom, go to second-to-last user message
   (since last is already visible), not to the last one
3. Add scroll awareness: Detect manual scrolling and adjust navigation
   state accordingly, so buttons adapt to current scroll position

philipperolet committed:
fix
